### PR TITLE
Add macos build test to modmesh_viewer.yml

### PR DESCRIPTION
**Pull request by minchuncho**
1. Specify the version of mac os: 12
2. Align the commands with the ones in https://github.com/solvcon/modmesh/blob/9eef4aaf004cbf86478368714e13493d34e0d59a/.github/workflows/modmesh_pytest.yml#L48

### CODE REVIEW

**Comment by minchuncho** [comment]
My old version of CI is as follows:
```
brew install wget make cmake libccd gcc python3 qt6
pip3 install -U numpy pytest flake8
```
Curious, I didn't write the following line but the old version still worked.
https://github.com/solvcon/modmesh/blob/9eef4aaf004cbf86478368714e13493d34e0d59a/.github/workflows/modmesh_pytest.yml#L52

I found that `setuptools` is a Python package development process library, i.e., enhanced `disutils`, while `disutils` is a Python standard library.
May I reach to this conclusion that if `setuptools` is not installed, then `disutils` is chosen by default so the old version still ran well?

File: .github/workflows/modmesh_viewer.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@ jobs:
 
     strategy:
         matrix:
-          os: [ubuntu-22.04]
+          os: [ubuntu-22.04, macos-latest]
```
Comment (yungyuc) [inline .github/workflows/modmesh_viewer.yml]: You have changed it to `macos-12` in #116 but here it goes back to `macos-latest`.  Perhaps an overlook?

**Review by yungyuc — CHANGES_REQUESTED** [review]
LGTM except the regression to os version alias.

**Comment by yungyuc** [comment]
> Curious, I didn't write the following line but the old version still worked.
> https://github.com/solvcon/modmesh/blob/9eef4aaf004cbf86478368714e13493d34e0d59a/.github/workflows/modmesh_pytest.yml#L52
> 
> I found that `setuptools` is a Python package development process library, i.e., enhanced `disutils`, while `disutils` is a Python standard library. May I reach to this conclusion that if `setuptools` is not installed, then `disutils` is chosen by default so the old version still ran well?

That is a reasonable guess, I think.  I do not know the answer either.  It probably does not matter ATM, but could be interesting to learn.

File: .github/workflows/modmesh_viewer.yml (hunk 1 above)
Comment (minchuncho) [reply .github/workflows/modmesh_viewer.yml]: Yes, it is an overlook and I have already fixed it.

Comment (yungyuc) [reply .github/workflows/modmesh_viewer.yml]: LGTM.  I will merge once the CI passes.

**Review by yungyuc — APPROVED** [review]

**Comment by yungyuc** [comment]
Thanks @minchuncho .  This addressed #111

**Pull request merged**